### DISCUSSION TITLE
Need Help for getting 3D Rotation of Map view

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#4176] [Q&A]
Hi , I want to get rotation view of Map in this sandbox : https://www.itwinjs.org/sandboxes/MyPOC/Dulciana-the-lime-angelfish-246

, can you help  to get the rotation of the Map view like this image :

**Comment by pmconne** [comment]
Please clarify your question. What are you trying to accomplish?

**Reply by Rahul-0108** [reply]
using the top right navigation cube , I have make the map view like this image which looks like 3d view,
I want to do that using code,
Please help

**Comment by pmconne** [comment]
If you want to capture everything about the current view and restore it later, you can use [ViewState.toProps](https://www.itwinjs.org/reference/core-frontend/views/viewstate/) to obtain the viewport's current view state as JSON to later be passed to [ViewState.createFromProps](https://www.itwinjs.org/reference/core-frontend/views/viewstate/createfrompropsstatic/). It only makes sense to do this in the context of the same iModel.

If you only care about the view frustum, save off [those paramters](https://www.itwinjs.org/learning/frontend/views/#viewstate-parameters) from the viewport's current view. Later you can pass them to [ViewState3d.lookAt](https://www.itwinjs.org/reference/core-frontend/views/viewstate3d/lookat/) to restore the frustum.
